--- a/data/cputable
+++ b/data/cputable
@@ -53,4 +53,5 @@
 sh4eb		sh4eb		sh4eb			32	big
 sparc		sparc		sparc			32	big
 sparc64		sparc64		sparc64			64	big
+sw64		sw_64		(sw_64|sw64).*		64	little
 tilegx		tilegx		tilegx			64	little
--- a/scripts/Dpkg/Vendor/Debian.pm
+++ b/scripts/Dpkg/Vendor/Debian.pm
@@ -249,14 +249,14 @@
 
     # Mask features that are not available on certain architectures.
     if ($os !~ /^(?:linux|kfreebsd|knetbsd|hurd)$/ or
-	$cpu =~ /^(?:hppa|avr32)$/) {
+	$cpu =~ /^(?:hppa|avr32|sw64)$/) {
 	# Disabled on non-(linux/kfreebsd/knetbsd/hurd).
-	# Disabled on hppa, avr32
+	# Disabled on hppa, avr32, sw64
 	#  (#574716).
 	$use_feature{hardening}{pie} = 0;
     }
-    if ($cpu =~ /^(?:ia64|alpha|hppa|nios2)$/ or $arch eq 'arm') {
-	# Stack protector disabled on ia64, alpha, hppa, nios2.
+    if ($cpu =~ /^(?:ia64|alpha|hppa|nios2|sw64)$/ or $arch eq 'arm') {
+	# Stack protector disabled on ia64, alpha, hppa, nios2, sw64.
 	#   "warning: -fstack-protector not supported for this target"
 	# Stack protector disabled on arm (ok on armel).
 	#   compiler supports it incorrectly (leads to SEGV)
--- a/scripts/t/Dpkg_Arch.t
+++ b/scripts/t/Dpkg_Arch.t
@@ -16,7 +16,7 @@
 use strict;
 use warnings;
 
-use Test::More tests => 18407;
+use Test::More tests => 18900;
 
 use_ok('Dpkg::Arch', qw(debarch_to_debtuple debarch_to_multiarch
                         debarch_eq debarch_is debarch_is_wildcard
@@ -28,7 +28,7 @@
                         get_host_gnu_type
                         get_valid_arches));
 
-my $KNOWN_ARCHES_TOTAL = 554;
+my $KNOWN_ARCHES_TOTAL = 569;
 my @valid_arches = get_valid_arches();
 
 sub get_valid_wildcards
